app/src/main/java/com/asuc/asucmobile/domain/main/LegendPopUpActivity.java

```diff
@@ -0,0 +1,93 @@
+package com.asuc.asucmobile.domain.main;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.os.PersistableBundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.app.DialogFragment;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.content.ContextCompat;
+import android.util.DisplayMetrics;
+import android.util.Pair;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import com.asuc.asucmobile.R;
+
+import java.util.ArrayList;
+
+public class LegendPopUpActivity extends DialogFragment {
+    public class LegendAdapter extends ArrayAdapter<Pair<Drawable, String>> {
+        public LegendAdapter(Context context, ArrayList<Pair<Drawable, String>> items) {
+            super(context, 0, items);
+        }
+
+        @NonNull
+        @Override
+        public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+            Pair<Drawable, String> pair = getItem(position);
+            if (convertView == null) {
+                convertView = LayoutInflater.from(getContext()).inflate(R.layout.legend_row, parent, false);
+            }
+            ImageView icon = (ImageView) convertView.findViewById(R.id.legend_icon);
+            TextView text = (TextView) convertView.findViewById(R.id.legend_text);
+            icon.setImageDrawable(pair.first);
+            text.setText(pair.second);
+
+            return convertView;
+        }
+    }
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        View v = inflater.inflate(R.layout.fragment_food_legend, container, false);
+
+        ListView legendList = (ListView) v.findViewById(R.id.legend_list);
+        ArrayList<Pair<Drawable, String>> items = new ArrayList<>();
+        items.add(new Pair(ContextCompat.getDrawable(this.getActivity(), R.drawable.milk), "Milk"));
```

Comment:
Would be cleaner if you defined these food type Strings somewhere else. See `values` folder

Reply:
@rustielin good point i will get on that